### PR TITLE
Display dimensions

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
PR to add a new Kotlin sample _"Display dimensions"_ in `Layers` category. This sample displays dimension features from a mobile map package.

## Links and Data
Sample Epic: `runtime/kotlin/issues/1939 `

## What To Review
-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

### CODE REVIEW

File: display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.lifecycleScope
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.mapping.ArcGISMap
```
Comment (prupani-7) [inline display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt:31]: there are some unused imports here

File: display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and load a mobile map package
+        val mobileMapPackage = MobileMapPackage("$provisionPath/Edinburgh_Pylon_Dimensions.mmpk")
```
Comment (prupani-7) [inline display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt:77]: we could move this string to string.xml

File: display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        activityMainBinding.mapView
+    }
+
+    private val optionsButton by lazy {
+        activityMainBinding.optionsButton
+    }
+
+    // keep an instance of the MapView's dimension layer
+    private var dimensionLayer: DimensionLayer? = null
+    // track if the layer is enabled
```
Comment (prupani-7) [inline display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt:63]: these would be separated by an empty line... Ran reformat Code in AS

**Review by prupani-7 — CHANGES_REQUESTED** [review]
@shubham7109 The sample works well! 

Added a few comments.

File: display-dimensions/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+![Image showing the Display Dimensions sample](display-dimensions.png)
+
+## Use case
+
+Dimensions show specific lengths or distances on a map. A dimension may indicate the length of a side of a building or land parcel, or the distance between two features, such as a fire hydrant and the corner of a building.
+
+## How to use the sample
+
+When the sample loads, it will automatically display the map containing dimension features from the mobile map package. The name of the dimension layer containing the dimension features is displayed in the controls box. Control the visibility of the dimension layer with the "Settings" button, and apply a definition expression to show dimensions of greater than or equal to 450m in length using the "Definition Expression" checkbox.
```
Comment (prupani-7) [inline display-dimensions/README.md:13]: "Dimension settings" button

**Comment by shubham7109** [comment]
Thanks @prupani-7  Added in the feedback

**Review by prupani-7 — APPROVED** [review]
LGTM!

File: display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+            // set up the dialog
+            AlertDialog.Builder(this).apply {
+                setView(dialogBinding.root)
+                setTitle("${getString(R.string.settings)}: ${dimensionLayer?.name}")
+
+                dimensionLayerSwitch.setOnCheckedChangeListener { _, isEnabled ->
+                    // set the visibility of the dimension layer
+                    dimensionLayer?.isVisible = isEnabled
+                    isDimensionLayerEnabled = isEnabled
```
Comment (hud10837) [inline display-dimensions/src/main/java/com/esri/arcgismaps/sample/displaydimensions/MainActivity.kt:119]: Could these just go in the apply blocks where you initialize `dimensionLayerSwitch`? Same for `definitionSwitch.setOnCheckedChangeListener`

File: display-dimensions/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            android:name=".MainActivity"
+            android:exported="true"
+            android:label="@string/app_name">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>
```
Comment (hud10837) [inline display-dimensions/src/main/AndroidManifest.xml:31]: Missing newline

**Review by hud10837 — APPROVED** [review]
Sample looks good, just one question but no need to re-review

**Pull request merged**